### PR TITLE
refactor(event): replace extra by context

### LINKED ISSUE #5: Put extra attributes in Sentry context instead of extra
First of all thanks for your slog plugins! I have a question about the Sentry plugin: You are currently putting all non-recognized log attributes into the `Extra` field of the Sentry `Event` struct in the `DefaultConverter`. However, looking at the Sentry documentation, it seems like they should go into `Contexts` instead, as `Extra` [has been deprecated](https://docs.sentry.io/platforms/javascript/enriching-events/context/).

To quote:
```
Additional Data is deprecated in favor of structured contexts.

Sentry used to support adding unstructured "Additional Data" via setExtra.
```

Was it a conscious choice to put it in `Extra`? (Groupings are also supported in `Context` where in `Extra` they are not)

### PR DESCRIPTION
**Pull request by samber**
Fix #5

### CODE REVIEW

File: converter.go [hunk 1]
```diff
@@ -94,9 +94,13 @@ func attrToSentryEvent(attr slog.Attr, event *sentry.Event) {
 			event.User.Data["request"] = anyValueToString(v)
 		}
 	} else if kind == slog.KindGroup {
-		event.Extra[attr.Key] = attrToMap(attr.Value.Group())
+		event.Contexts[attr.Key] = attrToMap(attr.Value.Group())
 	} else {
-		event.Extra[attr.Key] = attr.Value.Any()
+		// "context" should not be added to underlying context layers (see slog.KindGroup case).
+		if _, ok := event.Contexts["context"]; !ok {
+			event.Contexts["context"] = map[string]any{}
```
Comment (vansante) [inline converter.go:101]: I would use `make(map[string]any{}, 0)` here probably, optionally pre-allocating a few spots.

Comment (samber) [reply converter.go]: fixed

File: converter.go [hunk 2]
```diff
@@ -94,9 +94,13 @@ func attrToSentryEvent(attr slog.Attr, event *sentry.Event) {
 			event.User.Data["request"] = anyValueToString(v)
 		}
 	} else if kind == slog.KindGroup {
-		event.Extra[attr.Key] = attrToMap(attr.Value.Group())
+		event.Contexts[attr.Key] = attrToMap(attr.Value.Group())
 	} else {
-		event.Extra[attr.Key] = attr.Value.Any()
+		// "context" should not be added to underlying context layers (see slog.KindGroup case).
+		if _, ok := event.Contexts["context"]; !ok {
```
Comment (vansante) [inline converter.go:100]: You _could_ make "context" a public variable, optionally changeable by library users. (To stop overlapping Context names for instance)

Comment (samber) [reply converter.go:100]: Would you prefer "extra" ?

File: converter.go [hunk 3]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/getsentry/sentry-go"
 )
 
+const ContextKey = "extra"
```
Comment (vansante) [inline converter.go:15]: Making it a const means you cannot change it right? (Contrary to the readme)

Comment (samber) [reply converter.go:15]: fuuuuck 😜

Comment (samber) [reply converter.go:15]: see v1.1.1

**Pull request merged**